### ISSUE TITLE
Use PropTypes.elementType instead of PropTypes.func

### ISSUE THREAD

**Issue by gfpacheco** [GetStream/stream-chat-react-native#39]
The current props that expect a React element (rendered or not) require it to be either `node` or `func`:

```
PropTypes.oneOfType([PropTypes.node, PropTypes.func])
```

But a React component wrapped in `React.memo` (or [redux connected](https://github.com/reduxjs/react-redux/issues/1223#issuecomment-481308079)) is actually an `object`. I wonder if you could change their propType to expect `elementType` instead of `func`:

```
PropTypes.oneOfType([PropTypes.node, PropTypes.elementType])
```

That would fix a lot of warnings in my project.

**Comment by vishalnarkhede** [comment]
@gfpacheco thanks for bringing this up. We will look into it and get back to you as soon as possible :)

**Comment by vishalnarkhede** [comment]
Hey @gfpacheco , sorry for the delays. We have fixed this as part of [v0.5.0](https://github.com/GetStream/stream-chat-react-native/releases/tag/v0.5.0). Please let us know if it works for you :)
